### DISCUSSION TITLE
Can I add additional plugins?

### DISCUSSION THREAD

**Discussion by supersistence** [ObsidianPublisher/obsidian-github-publisher#144] [Q&A — Template]
Attempting to add mkdocs plugins to add citations and reading time.
How/can one add [mkdocs-timetoread](https://pypi.org/project/mkdocs-timetoread-plugin/0.0.4/#history) or other plugins?
I've added to mkdocs.yml and requirements.txt, but presume there is another install/load piece I am missing. 

Any advice?

**Comment by Mara-Li** [comment]
Hello!
Sorry for the long delay. If you don't make it through, you need to : 
- `pip install X` plugins you want
- Add it to your `requirements.txt`
- Add the plugin in your `mkdocs.yml` with their proper configuration (following the README)

**Reply by sensn** [reply]
Hello, I have a similar question, I wanted to add " markdown_extensions:
  - pymdownx.tabbed:
      alternate_style: true
"
To enable tabbed content. I added the above lines to mkdocs.yml. but now the GH page doesn't work. It builds but I get 404.
Do I also have to add something to requirements.txt?
Tabbed content would be a nice feature to have!

**Reply by Mara-Li** [reply]
Yes, you need to add to your requirements.txt 
(I hate this system of Python, that's why I use pipenv on my side). 

The requirements.txt tells to Python to install the package, and the mkdocs.yml tells to Mkdocs to enable it. If one is missing, you will lead to a crash.

**Reply by sensn** [reply]
Ok, thanks,

Looking for the version number I found this:
https://facelessuser.github.io/pymdown-extensions/extensions/blocks/plugins/tab/

For now I added pymdownx.tabbed==7.0 to requiremts. But it is the only entry starting with pymdown. 
I think it is still not working... Could you give an example of what to add to requirements?

**Reply by Mara-Li** [reply]
https://facelessuser.github.io/pymdown-extensions/installation/ → in requirements, just add pymdown-extensions
for the rest, i don't know for the moment. Pydown is intended to be used insidide plugin, or via markdown_extension mkdocs.yml

**Reply by sensn** [reply]
Hm, I use the Obsidian GH pages template, and I am editing the mkdocs and requirements.txt directly in place from GitHub repo. I do this from My phone.
Is it possible?

**Reply by Mara-Li** [reply]
What do you mean ? Editing a file inside github ? Yes, without problem!

**Reply by Mara-Li** [reply]
(Sorry if a sound mean, i want to help you quickly but im in a hurry :'D)

**Reply by Mara-Li** [reply]
So, my own mkdocs.yml have an example on how to use `pydownx` : https://github.com/Lisandra-dev/mara-li/blob/main/mkdocs.yml

edit this like that:
```yml
markdown_extensions:
	- pymdownx.blocks.tab
```

**Reply by Mara-Li** [reply]
Yes you can edit file in the github website on mobile or desktop. I already do that so don't worry ^^

**Reply by sensn** [reply]
Yes I do it but what do I need to add to requirements to add tabbed content. My repo is https://github.com/sensn/DigitalGarden/blob/main/requirements.txt

**Reply by sensn** [reply]
Alright thank you, I,lol have a look at your suggestions.

**Reply by Mara-Li** [reply]
To resume : 
- Add this to your `requirements.txt` : `pymdown-extensions`
- Edit your `mkdocs.yml` with: 
```yml
markdown_extensions:
  - pymdownx.blocks.tab
```

**Reply by sensn** [reply]
I have now a working build with
markdown_extensions:
  - pymdownx.tabbed:
      alternate_style: true
In mkdocs.yml and
pymdown-extensions==10.3
In requirements.txt

However I didn't get -pymdown.blocks.tab to work
I also tried adding -pymdown.blocks but build failed with "pymdown.blocks. Plugin not installed".
I think we somehow need to install blocks to use blocks.tab" but now don't know how.

So I cannot use the new syntax but o got tabbed content working. Thanks!

**Reply by Mara-Li** [reply]
Did you make a typo ? It's `pymdownx.blocks.tab` not `pymdown.blocks.tab`

**Comment by sensn** [comment]
If I do:

markdown_extensions:
  -pymdownm.blocks
  - pymdownx.blocks.tab



I get build error: 
ERROR   -  Config value 'markdown_extensions': Failed to load extension 'pymdownx.blocks'.
  File "/home/runner/work/DigitalGarden/DigitalGarden/venv/lib/python3.11/site-packages/mkdocs/config/config_options.py", line 1021, in run_validation
    md.registerExtensions((ext,), self.configdata)
  File "/home/runner/work/DigitalGarden/DigitalGarden/venv/lib/python3.11/site-packages/markdown/core.py", line 175, in registerExtensions
    ext = self.build_extension(ext, configs.get(ext, {}))

Aborted with 1 configuration errors!
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/DigitalGarden/DigitalGarden/venv/lib/python3.11/site-packages/markdown/core.py", line 236, in build_extension
    return module.makeExtension(**configs)
           ^^^^^^^^^^^^^^^^^^^^
AttributeError: Failed to initiate extension 'pymdownx.blocks': module 'pymdownx.blocks' has no attribute 'makeExtension'
Error: Process completed with exit code 1.

**Reply by Mara-Li** [reply]
I don't know, i think the best is to contact https://github.com/facelessuser/pymdown-extensions
